File: back-end/client/models/Voice.py
from datetime import datetime
from pydantic import BaseModel


class Voice(BaseModel):
    id: int | None = None
    remote_id: int | None = None
    name: str
    length: int
    used_times: int = 0
    hash_content: str
    created_at: datetime = datetime.now()
    updated_at: datetime = datetime.now()

File: front-end/src/model/Voice.ts
export interface Voice {
  id: number;
  remote_id: number;
  name: string;
  alias: string | null;
  length: number;
  used_times: number;
  hash_content: string;
  created_at: string;
  updated_at: string;
}

export interface ImportFileResponse {
  path: string;
  status: string;
  voice: Voice;
  error: string;
}